xarray/convert.py

```diff
@@ -49,3 +58,81 @@ def set_cdms2_attrs(var, attrs):
     cdms2_var = cdms2.createVariable(var.values, axes=axes, id=dataarray.name)
     set_cdms2_attrs(cdms2_var, var.attrs)
     return cdms2_var
+
+
+# TODO: Add converting bounds from xarray to Iris and back
+# TODO: Cell methods are not converted between Iris and xarray
+def to_iris(dataarray):
+    """Convert a DataArray into a Iris Cube
+    """
+    # Iris not a hard dependency
+    import iris
+    # iris.unit is deprecated in Iris v1.9
+    import cf_units
+
+    def check_attrs(attrs, keys):
+        return dict((k, v) for k, v in attrs.items() if k in keys)
+
+    def get_args(attrs):
+        _args = {'attributes': filter_attrs(attrs, iris_forbidden_keys)}
+        _args.update(check_attrs(attrs, ('standard_name', 'long_name',)))
+        _unit_args = check_attrs(coord.attrs, ('calendar',))
+        if attrs.has_key('units'):
+            _args['units'] = cf_units.Unit(attrs['units'], **_unit_args)
+        return _args
+
+    dim_coords = []
+    aux_coords = []
+
+    for coord_name in dataarray.coords:
+        coord = encode(dataarray.coords[coord_name])
+        coord_args = get_args(coord.attrs)
+        coord_args['var_name'] = coord_name
+        iris_coord = iris.coords.DimCoord(coord.values, **coord_args)
+        axis = None
+        if coord.dims:
+            axis = dataarray.get_axis_num(coord.dims)
+        if coord_name in dataarray.dims:
+            dim_coords.append((iris_coord, axis))
+        else:
+            aux_coords.append((iris_coord, axis))
+
+    args = get_args(dataarray.attrs)
+    args['var_name'] = dataarray.name
+    args['dim_coords_and_dims'] = dim_coords
+    args['aux_coords_and_dims'] = aux_coords
+
+    cube = iris.cube.Cube(dataarray.to_masked_array(), **args)
+    return cube
+
+
+def from_iris(cube):
+    """Convert a Iris cube into an DataArray
+    """
+    def get_attr(_obj):
+        attrs = {'standard_name': _obj.standard_name,
+                 'long_name': _obj.long_name}
+        if _obj.units.calendar:
+            attrs['calendar'] = _obj.units.calendar
+        if _obj.units.origin != '1':
+            attrs['units'] = _obj.units.origin
+        attrs.update(_obj.attributes)
+        return dict((k, v) for k, v in attrs.items() if v is not None)
+
+    name = cube.var_name
+    dims = [dim.var_name for dim in cube.dim_coords]
+    coords = OrderedDict()
+
+    for coord in cube.coords():
+        coord_attrs = get_attr(coord)
+        coord_dims = [cube.coords()[i].var_name for i in cube.coord_dims(coord)]
```

Comment:
Yes I should actually be able to use `dims` as a lookup here, which is a list of the dimension names.